### PR TITLE
C++: Mark everything in CodeDuplication.qll as deprecated

### PR DESCRIPTION
**Pull request by jketema**
Although we earlier added a comment to the classes in noting that they are deprecated, we did not properly mark the classes as actually being deprecated in https://github.com/github/codeql/pull/8036.

All predicates - except for 3 - depend on the classes being functional, which they no longer are, so mark those a deprecated as well. The three remaining predicates (`FunctionDeclarationEntry`, `numberOfSourceMethods`, and `whitelistedLineForDuplication`) seem to be helpers, and are likely not used when the library is not used, so mark those as deprecated as well.

### CODE REVIEW

File: cpp/ql/src/external/CodeDuplication.qll [hunk 1]
```diff
@@ -74,29 +74,37 @@ class DuplicateBlock extends Copy {
...
+deprecated class SimilarBlock extends Copy {
   override string toString() {
     result = "Similar code: " + this.sourceLines() + " almost duplicated lines."
   }
 }
 
-/** Gets a function with a body and a location. */
-FunctionDeclarationEntry sourceMethod() {
+/**
+ * DEPRECATED: The CodeDuplication library will be removed in a future release.
```
Comment (MathiasVP) [inline cpp/ql/src/external/CodeDuplication.qll:84]:
```suggestion
 * DEPRECATED: The `CodeDuplication` library will be removed in a future release.
```

File: cpp/ql/src/external/CodeDuplication.qll [hunk 2]
```diff
@@ -74,29 +74,37 @@ class DuplicateBlock extends Copy {
...
+deprecated FunctionDeclarationEntry sourceMethod() {
   result.isDefinition() and
   exists(result.getLocation()) and
   numlines(unresolveElement(result.getFunction()), _, _, _)
 }
 
-/** Gets the number of member functions in `c` with a body and a location. */
-int numberOfSourceMethods(Class c) {
+/**
+ * DEPRECATED: The CodeDuplication library will be removed in a future release.
```
Comment (MathiasVP) [inline cpp/ql/src/external/CodeDuplication.qll:95]:
```suggestion
 * DEPRECATED: The `CodeDuplication` library will be removed in a future release.
```

**Review by MathiasVP — DISMISSED** [review]
Thanks for doing this!

**Comment by jketema** [comment]
> Thanks for doing this!

Unfortunately CI is now failing because of this. It's failing on the query files in the same directory; which have been marked as deprecated for 17 months. I guess I could remove them.

**Comment by MathiasVP** [comment]
> Unfortunately CI is now failing because of this. It's failing on the query files in the same directory; which have been marked as deprecated for 17 months. I guess I could remove them.

Ah, I see. If it's indeed been deprecated for so long feel free to remove them!

**Review by MathiasVP — APPROVED** [review]
:shipit:

**Comment by jketema** [comment]
For completeness. The commit marking the queries as deprecated: https://github.com/github/codeql/commit/fc45b6cd3cee747044469669fc3680bf2ec79f64

**Comment by jketema** [comment]
Removing one more deprecated query made me touch [config/identical-files.json](https://github.com/github/codeql/pull/8366/files#diff-3abab5de80197372a3cc8d192e4e5abf1d026265b819d08b2dd1c406d691b9ff). Not sure whether this means other people now also need to review this.

**Review by MathiasVP — APPROVED** [review]

**Pull request merged**